server/server.go

```diff
@@ -126,7 +126,7 @@ type Server struct {
 	serverLoopWg     sync.WaitGroup
 
 	// for PD leader election.
-	member *member.Member
+	member *member.EmbeddedEtcdMember
```

Comment:
Could we not directly expose the underlying member type since we have abstracted an interface?

Reply:
I tried, but that need more refactor as pd server directly operates on s.member.Etcd() which can't be in the interface. If we let Sever keep etcd, it will introduce more change into pd server code and it isn't necessary for now.

Reply:
It mightn't be necessary for using the abstracted interface in server.go, because it depends on EmbeddedEtcdMember.Etcd() and pdpb.Member returned EmbeddedEtcdMember.Member()/GetLeader(). The abstracted interface can't expose embed etcd and pdpb.Member, because that's what we want to hide from underlying AllocatorManager and Local/Global TSO Allocator and why we define this Member interface. If using member.Member inteface in pd server.go, server struct needs to hold extra embedded etcd member variable and do type conversion everytime when the other compoments call Server.Member()/GetLeader() to get pdpb.Member info. So we mainly use the interface in the underlying AllocatorManager and Local/Global TSO Allocator.

Reply:
If so, can we just only define the functions which AllocatorManager will use?

Reply:
Yeah, if the interface is not wildly used to constrain the member itself, maybe it should be defined in the allocator rather than the `member` pkg.

Reply:
you're right. I'll move the Member interface into pkg/tso

Reply:
I like this suggestion very much.

Reply:
moved